containers/Ideas/Idea.js

```diff
@@ -162,88 +181,83 @@ export const Idea = (props) => {
+          processComment(data)
+          setSubmitPending(false)
+        } else if (statusCode === 403) {
+          return Promise.reject(data.detail)
+        } else if (statusCode === 400) {
+          const errorMsg = 'comment' in data
+            ? 'Comment: ' + data['comment']
+            : 'Bad request'
+          return Promise.reject(errorMsg)
         }
       })
+      .catch(error => {
+        setError(error)
+        setSubmitPending(false)
+      })
+  }
 
+  const getCommentPayload = (values, token) => {
+    return isEditing
+      ? [
+        editedComment.content_type,
+        editedComment.object_pk,
+        editedComment.id,
+        values,
+        token
+      ]
+      :
+      [
+        contentObjectOfComment.contentType,
+        contentObjectOfComment.pk,
+        values,
+        token
+      ]
   }
 
-  function handleCommentEdit(values) {
-    commentInputRef.current.blur()
-    AsyncStorage.getItem('authToken')
-      .then((token) => {
-        values.agreed_terms_of_use = true
-        return API.editComment(
-          editedComment.content_type,
-          editedComment.object_pk,
-          editedComment.id,
-          values,
-          token
-        )
-      })
-      .then((response) => {
-        const {statusCode, data} = response
-        if (statusCode == 200) {
-          toggleEditing()
-          if (data.content_type == data.comment_content_type) {
-            setCommentLastCommented(data.object_pk)
-          } else {
-            setCommentLastCommented(-1)
-          }
-          fetchComments(idea.content_type, idea.pk)
-          setContentObjectOfComment({'contentType': idea.content_type, 'pk': idea.pk})
-        } else {
-          const errorMessage = 'That did not work.'
-          let errorDetail
-          if (statusCode==403) {
-            errorDetail = data.detail
-          } else if (statusCode == 400) {
-            errorDetail = ('comment' in data ? ('Comment: ' + data['comment']) : 'Bad request')
-          }
-          Alert.alert(errorMessage, errorDetail, [{ text: 'Ok', onPress: ()=> {toggleEditing()} }])
-        }
-      })
+  const processComment = (data) => {
+    if (data.content_type == data.comment_content_type) {
+      setCommentLastCommented(data.object_pk)
+    } else {
+      setCommentLastCommented(-1)
+    }
+    fetchComments(idea.content_type, idea.pk)
+    setContentObjectOfComment({
+      'contentType': idea.content_type,
+      'pk': idea.pk
+    })
   }
 
   const handleCommentReply = (commentContentType, commentObjectPk) => {
-    setContentObjectOfComment({'contentType': commentContentType, 'pk': commentObjectPk})
+    setContentObjectOfComment({
+      'contentType': commentContentType,
+      'pk': commentObjectPk
+    })
     commentInputRef.current.focus()
   }
 
   function handleScroll(event) {
-    if (isScrolling && event.nativeEvent.contentOffset.y < 75.0) {
+    const { nativeEvent: { contentOffset: { y: offsetY } } } = event
```

Comment:
I find this somehow harder to read than just `const offsetY = event.nativeEvent.contentOffset.y. Is there any advantage to this style?

Reply:
i think just the matter of taste. Maybe because i am used to this notation. Will change it. Important to me is to "unpack" the variable `y` instead of having it in the condition.

Reply:
Fair, you can also leave it like this if you prefer it